### PR TITLE
Add BC formats.

### PR DESCRIPTION
**Pull request by Kangz**
+CC @Jiawei-Shao @litherum @kvark

### CODE REVIEW

**Comment by litherum** [comment]
I thought the whole point was to put these inside the extension, rather than in the main document.

**Comment by litherum** [comment]
Are IDL comments normative?

**Comment by kvark** [comment]
@litherum I don't think we talked enough about how extensions are going to be represented. In Khronos APIs, you can find the following documents describing the API:
  - core API for version X
  - each extension in a separate document, written against core API of version X
  - core API with all known extensions compatible with it

Is this close to what you are thinking?

(as a side discussion, extensions can be somewhat elegantly represented as PRs, which makes it clear what version they are written aganst, and how they affect the spec).

**Comment by Kangz** [comment]
IMHO extensions that we expect all vendors to implement could be in the main spec. They are like optional features in Vulkan ("KHR" extensions are essentially core too), or things gated by feature levels in Metal and D3D12. It makes sense to have them along the rest of the spec because they are the extensions people are most likely to use, and will go through this group's standardization process.

Vendor or multi-vendor extensions can be in separate document so as to not burden the main spec if someone comes up with a full GLES1 fixed-function style extension for example.

**Comment by kainino0x** [comment]
This confusion is the reason I've been talking about calling these optional features "capabilities" or "features" rather than "extensions". I think the word "extension" should be reserved for things which *are* specified outside of the main spec.

BC formats solidly fall into the category of things that should be in the main spec. It is present on all desktop platforms.

**Comment by kainino0x** [comment]
> Are IDL comments normative?

Not really, but there's no super-trivial way to organize this so that the blocks of formats are clearly written. It's better to have something here than nothing, but we should expect the comment to eventually get removed or made non-normative. Not everything we write into the spec right now can be shipping-quality, because editors haven't established the patterns yet.

File: spec/index.bs [hunk 1]
```diff
@@ -1659,6 +1659,22 @@ enum GPUTextureFormat {
     "depth32float",
     "depth24plus",
     "depth24plus-stencil8"
+
+    // BC compressed formats usable if "texture-compression-bc" is enabled.
```
Comment (Kangz) [inline spec/index.bs:1663]: nit: "is available and enabled"

Comment (jdashg) [inline spec/index.bs:1663]: "is available and enabled" would be even better!

Comment (kainino0x) [inline spec/index.bs:1663]:
```suggestion
    "depth24plus-stencil8",

    // BC compressed formats usable if "texture-compression-bc" is both
    // supported by the device/user agent and enabled in createDevice.
```

**Pull request merged**